model/map/src/main/java/org/keycloak/models/map/storage/MapStorageWithAuth.java

```diff
@@ -0,0 +1,27 @@
+package org.keycloak.models.map.storage;
+
+import org.keycloak.credential.CredentialModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.map.common.AbstractEntity;
+import org.keycloak.models.map.common.UpdatableEntity;
+
+/**
+ * Implementing this interface signals that the store can validate credentials.
+ * This will be implemented, for example, by a store that supports SPNEGO for Kerberos authentication.
+ *
+ * @author Alexander Schwartz
+ */
+public interface MapStorageWithAuth<V extends AbstractEntity & UpdatableEntity, M> extends MapStorage<V, M> {
```

Comment:
@pedroigor - from what I've seen, there will be two kinds of authentication: 

One is "on top" of the stores that works across stores, and one "built in" to a specific store.

The one "on top" is for example the `PasswordCredentialProvider`, as it will then put `CredentialModel` instances to the store. 

This wouldn't work for LDAP, as the authentication is "built in" to LDAP. And as LDAP won't be able to store all the other information we need to store about the user, it be used together with another store and the tree store. 

When retrieving the user, one would as the tree store to fetch information from JPA and LDAP and return it to the service layer. When validating credentials, one would pass them to the tree store that would then pass them on to the LDAP.

That's why we would need both, and we decided for this design.